### PR TITLE
Example: Show case how to add an extension point to a plugin

### PR DESCRIPTION
**Pull request by mckn**
Have dependency on canary because `9.5.0` npm packages not yet published

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,12 +4,13 @@ This repository contains example plugins to showcase different use cases.
...
+| Example                                                       | Description                                                                                       |
+| ------------------------------------------------------------- | ------------------------------------------------------------------------------------------------- |
+| [app-basic](examples/app-basic)                               | demonstrates how to build a basic app plugin that uses custom routing.                            |
+| [app-with-dashboards](examples/app-with-dashboards)           | demonstrates how to include pre-built dashboards in an app plugin.                                |
+| [app-with-backend](examples/app-with-backend)                 | demonstrates how to build an app plugin with its own backend.                                     |
+| [app-with-extensions](examples/app-with-extensions)           | demonstrates how to build an app plugin that extends the Grafana core ui.                         |
+| [app-with-extension-point](examples/app-with-extension-point) | demonstrates how to add an extension point in the plugin UI that can be extended by other plugins |
```
Comment (leventebalogh) [inline README.md:13]: **Nit:** should we maybe call this placement (to be coherent with how we call these in other places)?
```suggestion
| [app-with-extension-point](examples/app-with-extension-point) | demonstrates how to add an extension placement in the plugin UI that can be extended by other plugins |
```

File: examples/app-with-extension-point/README.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
+# Grafana App Plugin which adds a plugin UI extension point
+
+This example demonstrates how to add a plugin extension point that can be extended by other plugins. It works by bundling three different app plugins within the same bundle. First we have the `myorg-extensionpoint-app` that creates an extension point. Then we have the `src/plugins/myorg-a-app` and `src/plugins/myorg-b-app` that extends the `myorg-extensionpoint-app` UI with a link and a command.
```
Comment (leventebalogh) [inline examples/app-with-extension-point/README.md:3]: **Nit:** the same question as above.

File: examples/app-with-extension-point/README.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+## Guides in this example
+
+| **Example**                                                      | **Source**       |
+| ---------------------------------------------------------------- | ---------------- |
+| [How to add an extension point?](#how-to-add-an-extension-point) | [App.tsx#L12-18] |
+
+### How to add an extension point?
+
+**Example:** [App.tsx#L12-18]
```
Comment (leventebalogh) [inline examples/app-with-extension-point/README.md:15]: **Nit:** should we link the source? Or is this intended to be a foot-note?

File: examples/app-with-extension-point/README.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+The first thing you need to do is to define a placement for your extension point. A placement is a unique identifier that other plugins can reference when adding extensions to your extension point. It needs to start with `plugins/<your-plugin-id>/` followed by a relevent path for your plugin.
```
Comment (leventebalogh) [inline examples/app-with-extension-point/README.md:17]: **Nit:** maybe, just an idea.
```suggestion
The first thing you need to do is to define a placement for your extension point. A placement is a unique identifier that other plugins can reference when adding extensions to your extension point. It should be in the following format: `plugins/<your-plugin-id>/<placement-name>`, where:
- `<your-plugin-id>` is the full ID of your plugin, e.g. `"myorg-b-app"`
- `<placement-name>` is an identifier for the placement, unique inside your plugin, e.g. `"header"`
```

File: examples/app-with-extension-point/src/components/App/App.tsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+    const { extensions } = getPluginExtensions({
+      placement,
+      context,
+    });
+
+    return (
+      <div className="page-container" style={{ marginTop: '5%' }}>
+        <HorizontalGroup>
+          <span>Hello Grafana! This is the actions you can trigger from this plugin</span>
```
Comment (leventebalogh) [inline examples/app-with-extension-point/src/components/App/App.tsx:22]: **Nit**
```suggestion
          <span>Hello Grafana! These are the actions you can trigger from this plugin:</span>
```

**Review by leventebalogh — APPROVED** [review]
Nice one! 👏 

Overall it looks good to me, only left a few nits.

File: examples/app-with-extension-point/README.md (hunk 3 above)
Comment (mckn) [reply examples/app-with-extension-point/README.md:15]: I have no idea. Please help me here 🙈

**Pull request merged**